pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ZeroQueueSizeTest.java

```diff
@@ -305,17 +306,23 @@ public void testFailedZeroQueueSizeBatchMessage() throws PulsarClientException {
             producer.send(message.getBytes());
         }
 
-        try {
-            consumer.receiveAsync().handle((ok, e) -> {
-                if (e == null) {
-                    // as zero receiverQueueSize doesn't support batch message, must receive exception at callback.
-                    Assert.fail();
-                }
-                return null;
-            });
-        } finally {
-            consumer.close();
-        }
+        AtomicReference<Throwable> exceptionRef = new AtomicReference<>();
+
+        consumer.receiveAsync().handle((ok, e) -> {
+            // as zero receiverQueueSize doesn't support batch message, must receive exception at callback.
+            exceptionRef.set(e);
+            return null;
+        });
+        Awaitility.await().untilAsserted(() -> assertNotNull(exceptionRef.get()));
+        Awaitility.await().untilAsserted(() -> assertEquals(exceptionRef.get().getClass(),
+                PulsarClientException.InvalidMessageException.class));
+        Awaitility.await().untilAsserted(() -> assertEquals(exceptionRef.get().getMessage(),
```

Comment:
I think we don't need to await unitlAsserted here.

Reply:
That should be it. message consume is async.

Reply:
Sorry, I didn't express my point clearly. The lines 317 and 319 seem not to need to await unitlAsserted. Once we get the `exceptionRef` on line 316, the subsequent assert should not have to wait.

Reply:
Got.